media-sound/qpwgraph/qpwgraph-0.7.8.ebuild

```diff
@@ -20,27 +20,16 @@ fi
 LICENSE="GPL-2"
 SLOT="0"
 
-IUSE="alsa trayicon qt6 wayland"
+IUSE="alsa trayicon wayland"
```

Comment:
Neither `trayicon` nor `wayland` cause any additional dependencies. First of all, is that correct?

If they don't need additional deps, they shouldn't be there, unless there is something fundamental about them. If, e.g., enabling USE=wayland here would somehow degrade something in an X11 session, I would expect a metadata.xml description. If the tray icon can be disabled at runtime in a settings UI of the application, that would erase the usefulness of the USE flag completely imo.

Reply:
for wayland this is the code snippet depending on the cmake flag
```c++
int main ( int argc, char *argv[] )
{
	Q_INIT_RESOURCE(qpwgraph);
#if defined(Q_OS_LINUX) && !defined(CONFIG_WAYLAND)
	::setenv("QT_QPA_PLATFORM", "xcb", 0);
#endif
	qpwgraph_application app(argc, argv);
```
not sure if it's even needed, since it just sets/overrides an env var ?

For `trayicon`, the code depends on the define it toggles, so nothing we can do
```c++
#ifdef CONFIG_SYSTEM_TRAY
	m_settings->beginGroup(SystemTrayGroup);
	m_settings->setValue(SystemTrayEnabledKey, m_systray_enabled);
	m_settings->setValue(SystemTrayQueryCloseKey, m_systray_queryclose);
	m_settings->endGroup();
#endif
```

Reply:
I'd say it should be forced on given this code bit could be annoying for someone using neither X nor wayland (aka forced into xcb), not that we have many such users.

fwiw they do call the wayland support "experimental" for some reason which may mean it's better off, but with USE=wayland it'd be enabled for all wayland users anyway.

Reply:
I think I agree with you with keeping it enabled so that override doesn't happen.

Maybe it's best to patch that code snippet out, so I notice when doesn't apply anymore? Or maybe open a issue/PR upstream to remove it.

Reply:
I'd imagine they'll remove the wayland option then as it'll serve no other purpose, and should notice from either diff'ing cmake files on bump or the warning about the option being unused.

Reply:
Opened an issue https://gitlab.freedesktop.org/rncbc/qpwgraph/-/issues/104
I will change the `ebuild` to have `wayland` always enabled in cmake for now, and remove the useflag.

Reply:
> For `trayicon`, the code depends on the define it toggles, so nothing we can do
> 
> ```c++
> #ifdef CONFIG_SYSTEM_TRAY
> 	m_settings->beginGroup(SystemTrayGroup);
> 	m_settings->setValue(SystemTrayEnabledKey, m_systray_enabled);
> 	m_settings->setValue(SystemTrayQueryCloseKey, m_systray_queryclose);
> 	m_settings->endGroup();
> #endif
> ```

It just adds a default enabled "enable system tray icon" in help menu, so this should not be exposed as USE flag at all.

Reply:
@AdelKS ping - do you have any other reason to keep that flag?

Reply:
Yes, I'm repeating myself: You're not saving any dependency, and the feature is runtime configurable - so right now the IUSE is by definition not justified.